src/jit/lir.cpp

```diff
@@ -709,11 +665,102 @@ void LIR::Range::InsertBefore(GenTree* insertionPoint, GenTree* node1, GenTree*
 //
 void LIR::Range::InsertBefore(GenTree* insertionPoint, GenTree* node1, GenTree* node2, GenTree* node3, GenTree* node4)
 {
-    // TODO: Implement this directly, to avoid multiple updates to insertionPoint->gtPrev->gtNext.
-    InsertBefore(insertionPoint, node1);
-    InsertBefore(insertionPoint, node2);
-    InsertBefore(insertionPoint, node3);
-    InsertBefore(insertionPoint, node4);
+    assert(node1 != nullptr);
+    assert(node2 != nullptr);
+    assert(node3 != nullptr);
+    assert(node4 != nullptr);
+
+    assert(node1->gtNext == nullptr);
+    assert(node1->gtPrev == nullptr);
+    assert(node2->gtNext == nullptr);
+    assert(node2->gtPrev == nullptr);
+    assert(node3->gtNext == nullptr);
+    assert(node3->gtPrev == nullptr);
+    assert(node4->gtNext == nullptr);
+    assert(node4->gtPrev == nullptr);
+
+    node1->gtNext = node2;
+    node2->gtNext = node3;
+    node3->gtNext = node4;
+
+    node4->gtPrev = node3;
+    node3->gtPrev = node2;
+    node2->gtPrev = node1;
```

Comment:
I'm finding the batched gtNext/gtPrev updates a bit hard to read. Can we simply pair the updates for each  node? i.e.

```
node1->gtNext = node2;

node2->gtPrev = node1;
node2->gtNext = node3;

node3->gtPrev = node2;
node3->gtNext = node4;

node4->gtPrev = node3;
```